### ISSUE TITLE
Allow to choose translation table for prodigal when running anvi-gen-contigs-database

### ISSUE THREAD

**Issue by ShaiberAlon** [merenlab/anvio#1074]
Some bacteria use alternative translation. To support this, `prodigal` has this:

```
         -g:  Specify a translation table to use (default 11).
```

It would be nice to be able to specify that to `anvi-gen-contigs-database`.

**Comment by meren** [comment]
Are there example translation tables?

**Comment by ShaiberAlon** [comment]
Yes, for example the genome in this study:

https://www.ncbi.nlm.nih.gov/genome/16661

Is to be processed with table 25.

So it would be:

```
prodigal -g 25 .....
```

Here is the genome if you want to try:
```
wget ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/350/285/GCA_000350285.1_OR1/GCA_000350285.1_OR1_genomic.fna.gz ./
```

If we want to go crazy we can also compare the gene calls with the ones they have on genbank:
ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000/350/285/GCA_000350285.1_OR1

**Comment by meren** [comment]
Yes, I don't see why we wouldn't add this to contigs database. Although I don't like the idea of adding a parameter to an anvi'o program to be passed to another program (since gene caller should be modular, although we seem to have failed to design the contigs db module to be able to support that).

**Comment by ShaiberAlon** [comment]
>Yes, I don't see why we wouldn't add this to contigs database. 

Thank you!

>Although I don't like the idea of adding a parameter to an anvi'o program to be passed to another program

I understand.

**Comment by meren** [comment]
Please take a look and let me know if it works as expected, alon :)

**Comment by ShaiberAlon** [comment]
@meren, I have been experimenting with this, and was getting weird things.

I documented a bunch of details below, but here I will give you the bottom line:
I think that when `--prodigal-translation-table` then prodigal should be used with `-p single`, because anvio always uses prodigal with `-p meta`, but then prodigal ignores the translation table (There is an opened issue on the github repo for prodigal: https://github.com/hyattpd/Prodigal/issues/19).

The whole point of using translation table 25 is that `TGA` (AKA as the "opal stop codon") instead of being treated as a stop codon needs to be treated as a glycine (`G`).

So I ran `anvi-gen-contigs-database` with `--prodigal-translation-table 25` for a `GN02` [genome from NCBI](https://www.ncbi.nlm.nih.gov/assembly/GCF_002761215.1), and I'm seeing weird things, **most of it seems to be NOT anvi'o's fault**. What I noticed is that whether I use `--prodigal-translation-table 25` or not it doesn't change anything. So I decided to play with prodigal manually.

I ran prodigal manually, and I see that the parameter `-p meta` makes it so that the translation table is ignored. And since anvi'o always uses prodigal with `-p meta` then it always ignores the `--prodigal-translation-table`

So I ran prodigal without the `-p meta` (i.e. `-p single`, which is the default of prodigal) to compare the default translation table to translation table 25, and sadly I get two different results and both are different from the results of running prodigal with `-p meta`.

So overall I have four different gene calls:
1. gene calls produced with `-p meta` (translation table doesn't matter)
2. gene calls produced with `-p single` and `-g 11` (11 is the default translation table)
3. gene calls produced with `-p single` and `-g 25`
4. gene calls from NCBI (as produced by `anvi-script-process-genbank-metadata`) - this one was also different compared to all the rest

On the bright side (-ish) by doing a comparison by eye, I saw that only the table created by prodigal with `-g 11 -p single` was very very different from all the other three. The other three tables were predominantly similar, but with some differences (i.e. most genes were called in the same location), but there were a slightly different number of genes. But the problem is that the proteins that were produced with `-p meta` assign TGA to tryptophan (`W`), instead of glycine (`G`) (the reason is that in `-p meta`, prodigal tries translation table 11 on each sequence, and if the genes are too short then it defaults to translation table 4 (Mycoplasma), which translates TGA to `W`).
